pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/BucketDelayedDeliveryTrackerFactory.java

```diff
@@ -85,6 +88,9 @@ public DelayedDeliveryTracker newTracker(PersistentDispatcherMultipleConsumers d
      */
     public CompletableFuture<Void> cleanResidualSnapshots(ManagedCursor cursor) {
         Map<String, String> cursorProperties = cursor.getCursorProperties();
+        if (MapUtils.isEmpty(cursorProperties)) {
```

Comment:
I wonder which place introduce the `null` cursorProperties. And if so, should all the place check `cursorProperties` is empty or not:)?

Reply:
> I wonder which place introduce the null cursorProperties. And if so, should all the place check cursorProperties is empty or not:)?

Good suggestion. Fixed